engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -261,11 +280,290 @@ public void shouldActivateCompensationHandlerForEndEvent() {
+            tuple(
+                BpmnElementType.BOUNDARY_EVENT,
+                BpmnEventType.COMPENSATION,
+                ProcessInstanceIntent.ELEMENT_COMPLETED,
+                "CompensationBoundaryEvent"),
+            tuple(
+                BpmnElementType.USER_TASK,
+                BpmnEventType.COMPENSATION,
+                ProcessInstanceIntent.ELEMENT_ACTIVATING,
+                "CompensationHandler"),
+            tuple(
+                BpmnElementType.USER_TASK,
+                BpmnEventType.COMPENSATION,
+                ProcessInstanceIntent.ELEMENT_ACTIVATED,
+                "CompensationHandler"),
+            tuple(
+                BpmnElementType.USER_TASK,
+                BpmnEventType.COMPENSATION,
+                ProcessInstanceIntent.ELEMENT_COMPLETED,
+                "CompensationHandler"),
+            tuple(
+                BpmnElementType.END_EVENT,
+                BpmnEventType.COMPENSATION,
+                ProcessInstanceIntent.ELEMENT_COMPLETED,
+                "CompensationEndEvent"));
+  }
+
+  @Test
+  public void shouldActivateAndCompleteMultipleCompensationHandlerForThrowEvent() {
+    // given
+    final var process =
+        createModelFromClasspathResource("/compensation/multiple-compensation-throw-event.bpmn");
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE
+        .job()
+        .withKey(
+            RecordingExporter.jobRecords(JobIntent.CREATED)
+                .withElementId("ActivityToCompensate")
+                .getFirst()
+                .getKey())
+        .complete();
+
+    ENGINE
+        .job()
+        .withKey(
+            RecordingExporter.jobRecords(JobIntent.CREATED)
+                .withElementId("ActivityToCompensate2")
+                .getFirst()
+                .getKey())
+        .complete();
+
+    final long jobKeyOfCompensationHandler =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withElementId("CompensationHandler")
+            .getFirst()
+            .getKey();
+
+    ENGINE.job().withKey(jobKeyOfCompensationHandler).complete();
+
+    final long jobKeyOfCompensationHandler2 =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withElementId("CompensationHandler2")
+            .getFirst()
+            .getKey();
+
+    ENGINE.job().withKey(jobKeyOfCompensationHandler2).complete();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .extracting(
+            r -> r.getValue().getBpmnElementType(),
+            r -> r.getValue().getBpmnEventType(),
+            Record::getIntent,
+            r -> r.getValue().getElementId())
+        .containsSubsequence(
+            tuple(
```

Comment:
Hey @saig0 there are some test failures related to these assertions, I wasn't able to find the issue. I'm still on it

Reply:
✅